ledger/eval.go

```diff
@@ -762,6 +763,18 @@ func (eval *BlockEvaluator) endOfBlock() error {
 		} else {
 			eval.block.TxnCounter = 0
 		}
+
+		if eval.proto.CompactCertRounds > 0 && eval.block.Round()%basics.Round(eval.proto.CompactCertRounds) == 0 {
```

Comment:
consider changing the CompactCertVoters(loopback) so that it have a similar signature to something like:
```
err := CompactCertVoters(basics.Round, *CompactCertVoters, *CompactCertVotersTotal)
```

so that you could collapse the logic you’ve added to finalValidation and endOfBlock into a single location.

Reply:
The reason the `Ledger.CompactCertVoters()` function returns the entire `VotersForRound` is that I need other parts from that `VotersForRound` in the rest of the code.  Here's the complete compactcert implementation, if it's helpful, and the other place that I need to use `Ledger.CompactCertVoters()`: https://github.com/zeldovich/go-algorand/blob/compactcert/compactcert/builder.go#L48

But I agree it'd be good to factor out this common logic.  Pushed that change.